types/event.d.ts

```diff
@@ -310,10 +310,7 @@ interface PlayerChangedRet {
   // determine which job was passed in with the event and explicitly use that JobDetail
   // Potentially add the job to the jobDetail passed back from the C# plugin, and use
   // that information to decide the type
-  jobDetail: JobDetail['PLD'] & JobDetail['WAR'] & JobDetail['DRK'] & JobDetail['GNB'] & JobDetail['WHM'] &
-    JobDetail['SCH'] & JobDetail['AST'] & JobDetail['PGL'] & JobDetail['MNK'] & JobDetail['DRG'] &
-    JobDetail['NIN'] & JobDetail['SAM'] & JobDetail['BRD'] & JobDetail['MCH'] & JobDetail['DNC'] &
-    JobDetail['THM'] & JobDetail['BLM'] & JobDetail['ACN'] & JobDetail['SMN'] & JobDetail['RDM'];
+  jobDetail: JobDetail[T];
```

Comment:
You can delete my long-winded comment above that says "I'll defer to a wizard," as the wizard has arrived.

Reply:
I'm hardly a wizard, just an acolyte who found the right spellbook in the library 😉

Reply:
If T of `T extends Job` is not the key in `JobDetail`, would it be `never` or something?